attributes/default.rb

```diff
@@ -0,0 +1,24 @@
+#
+# Author:: Salim Afiune <afiune@chef.io>
+# Copyright (c) 2015, Chef Software, Inc. <legal@chef.io>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Custom repository setup recipe
+#
+# When the user specifies this attribute chef-ingredient will not configure
+# our default packagecloud Chef repositories and instead it will include the
+# custom recipe. This will eliminate the hard dependency to the internet.
+#
+default['chef-ingredient']['custom_repo_setup_recipe'] = nil
```

Comment:
Awesome! I just think that if I do that I will have the awful: 
```
NoMethodError
-------------
undefined method `[]' for nil:NilClass
```

Why not better:
```
default['chef-ingredient'] = {}
```